cdi/embedded/src/main/java/org/infinispan/cdi/embedded/DelegatingDistributedTaskLifecycle.java

```diff
@@ -61,14 +60,31 @@ public DelegatingDistributedTaskLifecycle() {
             BeanManager bm = CDIHelper.getBeanManager();
             if (bm == null)
                return;
-            Class<Callable<T>> clazz = (Class<Callable<T>>) task.getClass();
-            AnnotatedType<Callable<T>> type = bm.createAnnotatedType(clazz);
-            InjectionTarget<Callable<T>> it = bm.createInjectionTarget(type);
-            it.preDestroy(task);
-            it.dispose(task);
+            postInject(bm, task);
+            if(task instanceof DefaultExecutorService.RunnableAdapter)
+            {
```

Comment:
Could you please use proper [formatter](https://github.com/infinispan/infinispan/tree/master/ide-settings)?